integration-tests/src/test/java/com/datastax/oss/driver/core/compression/DirectCompressionIT.java

```diff
@@ -74,9 +74,10 @@ public static void setup() {
   @Test
   public void should_execute_queries_with_snappy_compression() throws Exception {
     Assume.assumeTrue(
-        "Snappy is not supported in OSS C* 4.0+ with protocol v5",
-        CCM_RULE.getDseVersion().isPresent()
-            || CCM_RULE.getCassandraVersion().nextStable().compareTo(Version.V4_0_0) < 0);
+        "Snappy is not supported in OSS C* 4.0+ with protocol v5 and DSE before 6.0",
+        (!CCM_RULE.getDseVersion().isPresent()
+                && CCM_RULE.getCassandraVersion().nextStable().compareTo(Version.V4_0_0) < 0)
+            || CCM_RULE.getDseVersion().get().nextStable().compareTo(Version.V6_0_0) >= 0);
```

Comment:
I don't think these two cases are identical, are they?

The old code test says to execute this test case under two conditions:

1. We're dealing with DSE (thus getDseVersion().isPresent())
2. We're not dealing with DSE and we have a C* version less than 4.0

If I'm understanding things we now want to change that logic to the following:

1. We're dealing with DSE and we have a DSE version less than 6.0
2. We're not dealing with DSE and we have a C* version less than 4.0

The most explicit way to express that is something like:

```
(CCM_RULE.getDseVersion().isPresent() && CCM_RULE.getDseVersion().get().nextStable().compareTo(Version.V6_0_0) >= 0)) || (!CCM_RULE.getDseVersion().isPresent() && CCM_RULE.getCassandraVersion().nextStable().compareTo(Version.V4_0_0) < 0))
```

AND. has a higher precedence than OR for boolean ops so as written this test will fail anytime DSE version is present... which means it'll fail incorrectly for DSE < 6.0.

Reply:
No matter how the logic looks in the end I'd argue it's a good idea to use parens in a logical expression as complicated as this one is going to be.  Even if it isn't necessary to get the logic right it still makes the code much easier for others to read after the fact.

Reply:
Fix the logical expression. Good catch!

Reply:
Strange, this test started passing. Rolled back this change.